src/Tutorials/Unit-Testing-Grains.md

```diff
@@ -5,81 +5,159 @@ title: Unit Testing Grains
 
 # Unit Testing Grains
 
-This tutorial shows you how to write unit tests for your grains to make sure they are behaving correctly.
-For a distributed application written in Orleans, you'll need load tests and integration tests as well but here we'll only focus on unit tests.
+This tutorial shows how to unit test your grains to make sure they behave correctly.
+There are two main ways to unit test your grains, and the method you choose will depend on the type of functionality you are testing.
+You can use a mocking framework like [Moq](https://github.com/moq/moq) to mock parts of the Orleans runtime that your grain interacts with, or you can use the  `Microsoft.Orleans.TestingHost` NuGet package to create test silos for your grains.
 
-Orleans makes it possible to mock many of its parts ([example](https://github.com/dotnet/orleans/tree/master/Samples/UnitTesting.Minimal)) but here we only focus on simply running grains in test silos.
+## Using Mocks
```

Comment:
Even though it's nice this is doable, I would prioritize TestCluster higher (and hence put it first in the docs).
Also with this approach different people will want to use different mocking frameworks (I love Moq btw). I hope that at some point we put more effort on having a better experience without requiring to mock out complex parts of Orleans yourself (but in the meantime, you might want to link to https://github.com/OrleansContrib/OrleansTestKit).